anta/tests/bfd.py

```diff
@@ -341,16 +311,10 @@ def test(self) -> None:
 
             # Check if BFD peer configured
             if not bfd_output:
-                failures[peer] = {vrf: "Not Configured"}
+                self.result.is_failure(f"{bfd_peer} - Not found")
                 continue
 
             # Check registered protocols
             difference = set(protocols) - set(get_value(bfd_output, "peerStatsDetail.apps"))
-
             if difference:
-                failures[peer] = {vrf: sorted(difference)}
-
-        if not failures:
-            self.result.is_success()
-        else:
-            self.result.is_failure(f"The following BFD peers are not configured or have non-registered protocol(s):\n{failures}")
+                self.result.is_failure(f"{bfd_peer} - `{','.join(difference)}` routing protocols not configured")
```

Comment:
`difference` is a set so please confirm the order does not matter, otherwise use `sorted`

Reply:
Updated. Added sorted. Thanks!!